identity/resource_service_principal.go

```diff
@@ -116,6 +116,15 @@ func ResourceServicePrincipal() *schema.Resource {
 			if err := common.DataToStructPointer(d, servicePrincipalSchema, &sp); err != nil {
 				return err
 			}
+			if c.IsAzure() && sp.ApplicationID == "" {
```

Comment:
should we have some check for GCP - throw the error that we don't support SPs there?

Reply:
can you create internal ticket for backend to return sane error messages? :)